### PR TITLE
Convert to ES module

### PR DESCRIPTION
**Pull request by fisker**

### CODE REVIEW

File: index.js
Comment (josephfrazier) [inline index.js:1]: The changes to the export syntax would mean that `require('editorconfig-to-prettier')` will no longer work, right? At least, here's what I get locally on this branch:

```
~/workspace/editorconfig-to-prettier$ node
Welcome to Node.js v16.19.0.
Type ".help" for more information.
> require('./')
Uncaught:
Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/josephfrazier/workspace/editorconfig-to-prettier/index.js not supported.
Instead change the require of index.js in null to a dynamic import() which is available in all CommonJS modules.
    at __node_internal_captureLargerStackTrace (node:internal/errors:478:5)
    at new NodeError (node:internal/errors:387:5)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1192:19)
    at Module.load (node:internal/modules/cjs/loader:1043:32)
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)
    at Module.require (node:internal/modules/cjs/loader:1067:19)
    at require (node:internal/modules/cjs/helpers:103:18) {
  code: 'ERR_REQUIRE_ESM'
}
>
```

**Review by josephfrazier — COMMENTED** [review]
Thanks for this, @fisker! Due to the `require` vs `import` change mentioned inline, I think this should be considered a breaking change. Does that make sense to you? I have no problem with bumping to v2, just wanted to get a sense of how folks are treating this type of change.

**Comment by fisker** [comment]
Yes, it's a breaking change.

**Comment by josephfrazier** [comment]
Thanks for confirming. Can you describe the benefits of making this package be an ES module? Given the small size and simplicity of the package (e.g. there's only one export), it's not clear to me that breaking `require('editorconfig-to-prettier')` is worth it, after thinking about it a little more.

I'm definitely still open to the change, just want to make sure I understand it.

**Comment by fisker** [comment]
As I understand, this package is made for Prettier to use, Prettier is now ES Module.

Do we have other users?

**Comment by josephfrazier** [comment]
apologies for not getting back to you on this earlier, but I responded in https://github.com/josephfrazier/editorconfig-to-prettier/issues/24#issuecomment-1718405985

**Comment by josephfrazier** [comment]
Fixed in https://github.com/prettier/prettier/pull/15394

**Pull request closed without merging**